src/app/devices/components/DeviceHeaderForms/AddDeviceForm/AddDeviceForm.tsx

```diff
@@ -38,17 +42,78 @@ const AddDeviceSchema = Yup.object().shape({
           .matches(MAC_ADDRESS_REGEX, "Invalid MAC address")
```

Comment:
I'd suggest to extract to extract Interfaces validation to `InterfaceSchema`, `AddDeviceSchema` will be much easier to reason about.

```
interfaces: Yup.array()
    .of(InterfaceSchema)
```